Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -345,6 +345,12 @@ class CustomBuildTaskTerminal implements Pseudoterminal {
     }
 
     async open(_initialDimensions: TerminalDimensions | undefined): Promise<void> {
+        const editor: TextEditor | undefined = window.activeTextEditor;
+        if (editor && !util.fileIsCOrCppSource(editor.document.fileName)) {
+            this.writeEmitter.fire(localize("cannot.build.non.cpp", 'Cannot build and debug because the active file is not a C or C++ source file.') + this.endOfLine);
+            this.closeEmitter.fire(-1);
+            return Promise.resolve();
```

Comment:
This function is `async` so it should just `return;`  There is no need to call `Promise.resolve()`